src/displayapp/screens/settings/SettingWakeUp.cpp

```diff
@@ -85,27 +85,39 @@ bool SettingWakeUp::Refresh() {
 }
 
 void SettingWakeUp::UpdateSelected(lv_obj_t* object, lv_event_t event) {
-  if (event == LV_EVENT_VALUE_CHANGED) {
+  if (event == LV_EVENT_VALUE_CHANGED and !ignoreEvent) {
+    // Ignore all intermediate events that will be generated by checkboxes
+    // while we modify their state in this event handler.
+    ignoreEvent = true;
+    // lv_checkbox_set_checked(cbOption[0], true);
     for (int i = 0; i < optionsTotal; i++) {
       if (object == cbOption[i]) {
-        lv_checkbox_set_checked(cbOption[i], true);
-
+        auto& currentCheckbox = cbOption[i];
+        auto currentCheckboxState = lv_checkbox_is_checked(currentCheckbox);
+        using WakeUpMode = Controllers::Settings::WakeUpMode;
         if (i == 0) {
-          settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::None);
-        };
-        if (i == 1) {
-          settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::SingleTap);
-        };
-        if (i == 2) {
-          settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::DoubleTap);
-        };
-        if (i == 3) {
-          settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::RaiseWrist);
-        };
-
-      } else {
-        lv_checkbox_set_checked(cbOption[i], false);
+          for (auto& cb : cbOption) {
+            lv_checkbox_set_checked(cb, false);
+          }
+          settingsController.resetWakeUpModes();
+        } else if (i == 1 + static_cast<size_t>(WakeUpMode::SingleTap)) { // minus 1 because first checkbox is the `None` checkbox
+          // SingleTap and DoubleTap checkboxes cannot be both checked at the same time.
+          if (!currentCheckboxState)
+            lv_checkbox_set_checked(cbOption[1 + static_cast<size_t>(WakeUpMode::DoubleTap)], false);
+          settingsController.setWakeUpMode(WakeUpMode::SingleTap, currentCheckboxState);
+        } else if (i == 1 + static_cast<size_t>(WakeUpMode::DoubleTap)) {
+          // SingleTap and DoubleTap checkboxes cannot be both checked at the same time.
+          if (!currentCheckboxState)
+            lv_checkbox_set_checked(cbOption[1 + static_cast<size_t>(WakeUpMode::SingleTap)], false);
+          settingsController.setWakeUpMode(WakeUpMode::DoubleTap, currentCheckboxState);
+        } else if (i == 1 + static_cast<size_t>(WakeUpMode::RaiseWrist)) {
+          settingsController.setWakeUpMode(WakeUpMode::RaiseWrist, currentCheckboxState);
+        }
```

Comment:
Up to you, but in the same way that the `None` enumeration wasn't really needed, I don't think that the None checkbox is really needed either since selecting it is effectively the same as just unchecking all of the other options.

Reply:
I'd prefer not to change the UI too much, and a "None" checkbox can be useful to turn all other modes at once.